WordPress/gutenberg
# REST API: Avoid resetting parent field in draft update

Related: #10753 

This pull request seeks to resolve an issue where if a draft page is assigned a parent, the parent will become unset the next time an autosave occurs. This occurs because while the revision's parent is typically assigned to the post for which the revision is to be created, in this flow a revision is not created and instead the post itself is updated.

**Testing instructions:**

Verify that an autosave for a page assigned a parent does not ~reset the parent value~ _send the `parent` value in its save request_.

1. Navigate to Pages > Add New
2. Set a Parent page
3. Add a title
4. Save the page
5. Reload
6. Change the title
7. Wait for autosave
8. Once autosave occurs, ~reload~ _observe in the Developer Tools Network tab that a `parent` value was not sent with the request_.

~Note that the parent field is still assigned to the value chosen in Step 2.~

Verify there are no other regressions in the behavior of autosave.

Ensure tests pass:

```
npm run test-unit
```

```
npm run test-unit-php
```

```
npm run test-e2e
```

**Follow-up Task:**

Needs Trac ticket / patch for core 5.0 branch if deemed an appropriate solution.

Related:
* https://core.trac.wordpress.org/ticket/43316
* #6257
